teps/0007-conditions-beta.md

```diff
@@ -0,0 +1,612 @@
+
+An example using a `CelRun` can be specified as shown below:
+```
+api: tekton.dev/v1alpha1
+kind: Pipeline
+metadata: 
+    name: generate-file
+spec:
+    params:
+      - name: branch
+        type: string
+    tasks: 
+      - name: create-file
+        taskRef: 
+          name: create-readme-file 
+      - name: check-file-exists 
+        when:
+          - name: file-exists
+            taskRef: 
+              name: file-exists
+          - name: branch-is-main
+            taskSpec: 
+              apiVersion: custom.dev/v1beta1
+              kind: CelRun
+              spec:
+                  eval: $(params.branch) == ‘main’
+        taskRef:
+          name: echo-file-exists
+        runAfter:
+          create-file
+```
+When the `Pipeline` executes, it’ll create a `CelRun` object, and we’ll provide a `controller` to interpret and execute the `CelRun`. We can also experiment with using other languages, like `bash`, `scriptmode`, `jsonpath`. 
+
+### Skipping
+
+Provide more flexibility when a `Guard` executes as `False` by adding a boolean flag to specify whether to terminate the whole `Task` branch, or to skip the guarded `Task` only and continue with the rest of the `Task` branch. 
+
+For example, when a `Task` is dependent on a couple of parent `Tasks`, and a user doesn’t want it to be skipped because of a guarded `Task` evaluating to `False`, they can specify the continueAfterSkip to allow for execution of the dependent `Task`. 
+
+A `Task` branch is made up of dependent `Tasks`, where there are two types of dependencies:
+- Resource dependency: based on resources needed from parent `Task`, which includes Workspaces, `Results` and Resources. 
+- Ordering dependency: based on runAfter which provides sequencing of `Tasks` when there may not be resource dependencies. 
+
+Users can specify a continueAfterSkip boolean, which defaults to `False`, and they can set it to true to allow for execution of the rest of the branch. 
+
+```
+api: tekton.dev/v1alpha1
+kind: Pipeline
+metadata: 
+    name: generate-file
+spec: 
+    tasks: 
+      - name: create-file                 # executed
+        taskRef: create-readme-file 
+      - name: echo-file-when-exists       # skipped
+        when:
+          - name: file-exists
+            taskRef: 
+              name: file-exists
+        continueAfterSkip: true 
+        taskRef:
+          name: echo-file-exists
+        runAfter:
+          - create-file
+      - name: echo-hello                 # executed
+        taskRef: echo-hello
+        runAfter:
+        - echo-file-when-exists 
+```
+When continueAfterSkip is set to `True`, subsequent `Tasks` based on ordering dependencies should execute and the subsequent `Tasks` based resource dependencies will have resource validation errors.
```

Comment:
im still leaning toward treating a case where you set continueAfterSkip on a Task to true and there are resource dependencies on that task as invalid and failing it at Pipeline creation time

Reply:
sounds good, updated the doc and will modify that in the proof of concept